### ISSUE TITLE
BUG: reading layers in geodataframe(gdb) with null geometries returns error

### ISSUE THREAD

**Issue by Brjax** [geopandas/geopandas#2380]
`gdf = gpd.read_file('geo.gdb/', layer='unit_table')`

In prior versions reading tables with empty geometries (None in the geometry field) was not an issue. I am now receiving the follow error after updating.

`File C:\ProgramData\Anaconda3\envs\geo_env\lib\site-packages\geopandas\io\file.py:208, in _read_file(filename, bbox, mask, rows, **kwargs)
    200 with fiona_env():
    201     with reader(path_or_bytes, **kwargs) as features:
    202 
    203         # In a future Fiona release the crs attribute of features will
    204         # no longer be a dict, but will behave like a dict. So this should
    205         # be forwards compatible
    206         crs = (
    207             features.crs["init"]
--> 208             if features.crs and "init" in features.crs
    209             else features.crs_wkt
    210         )
    212         # handle loading the bounding box
    213         if bbox is not None:

File C:\ProgramData\Anaconda3\envs\geo_env\lib\site-packages\fiona\collection.py:215, in Collection.crs(self)
    213 """Returns a Proj4 string."""
    214 if self._crs is None and self.session:
--> 215     self._crs = self.session.get_crs()
    216 return self._crs

File fiona\ogrext.pyx:742, in fiona.ogrext.Session.get_crs()

File fiona\_err.pyx:291, in fiona._err.exc_wrap_pointer()

CPLE_AppDefinedError: PROJ: proj_identify: SQLite error on SELECT key, value FROM metadata WHERE key IN ('DATABASE.LAYOUT.VERSION.MAJOR', 'DATABASE.LAYOUT.VERSION.MINOR'): no such table: metadata`

**Comment by martinfleis** [comment]
Thanks for the report @Brjax. Can you share some small `gdb` with None we can use to debug this?

**Comment by Brjax** [comment]
Looks like this was actually an issue with pyproj. It appears to have been affected when updating geopandas in Anaconda. 
I created a new pyproj environment :
`conda config --prepend channels conda-forge
conda config --set channel_priority strict
conda create -n pyproj_env pyproj
conda activate pyproj_env`

then set the proj environment to the directory with proj.db:
'os.environ['PROJ_LIB'] = 'C:\\...\\share\\proj\\'
